.github/workflows/code-coverage.yml

```diff
@@ -0,0 +1,45 @@
+name: "Code Coverage"
```

Comment:
Since we have pipelines in Azure DevOps, I was thinking that the code-coverage report could be something that was part of the DevOps pipeline.
You could make sure that the tasks that run the test also generate the coverage statistics and then generate a report from them.

What exactly does this workflow do ?  It runs the tests yet again ?

Reply:
Since all our other reports and analysis are done in GitHub/Actions (security analysis, dependency analysis, code style...), it seemed good to do our code-coverage also this way. We currently use Azure DevOps only for our CI/Release pipeline and as I understand it, it will also be (in the far future) be moved to GitHub. Another reason for me to already do this here. 
This workflow runs the Coverlet stuff after a test-run, that's correct. Like our CodeQL analysis also builds again. We will have a report here as artifact instead of going to Azure DevOps.

Reply:
Ok, makes sense.

But then again, why don't we move our DevOps pipelines for CI builds and releases also to GH actions ?  That's a different discussion and has nothing to do with this PR, but I think we should consider that.